core/modal_models/modes.c

```diff
@@ -558,4 +557,51 @@ void _lf_terminate_modal_reactors() {
     }
     _lf_unsused_suspended_events_head = NULL;
 }
+
+/**
+ * @brief This function accepts an array of ALL the mode_states and should add a reactor ptr into return_vec 
+ *  for all reactors that will do a transition
+ * 
+ *  A mode does a transition of either:
+ *  1) next_mode is set or
+ *  2) parent has next_mode set
+ * 
+ *  A preliminary implementation is simple:
+ *  1) Check if reactor has next_mode set. If so, add to return_vec 
+ *  2) If not, check if its parent has already been added to return_vec. If so, add it
+ *  
+ *  This assumes that the `states` argument is an array ordered hierarchically. It does a bunch of searching through
+ *  the return_vec, but this might be OK.   
+ * 
+ * @param states 
+ * @param states_size 
+ * @param return_vec 
+ * @return int 
+ */
+int _lf_mode_collect_transitioning_reactors(reactor_mode_state_t **states, int states_size, void * _return_vec) {
```

Comment:
If the return type (for _return_vec) is self_base_t*, then it won't need to be cast when used, maybe? Also, is the leading underscore in _return_vec meaningful?

Reply:
`_return_vec` because it is void*, when casting it to `vector_t` i name it `return_vec`. Made sense in my head, but open for another way of expressing it